locator/abstract_replication_strategy.cc

```diff
@@ -468,6 +470,32 @@ void effective_replication_map_factory::submit_background_work(future<> fut) {
     });
 }
 
+future<> global_effective_replication_map::get_keyspace_erms(sharded<replica::database>& sharded_db, std::string_view keyspace_name) {
+    return sharded_db.invoke_on(0, [this, &sharded_db, keyspace_name] (replica::database& db) -> future<> {
+        auto lk = co_await db.get_shared_token_metadata().get_lock();
+        auto erm = db.find_keyspace(keyspace_name).get_effective_replication_map();
+        auto ring_version = erm->get_token_metadata().get_ring_version();
+        _erms[0] = make_foreign(std::move(erm));
+        co_await coroutine::parallel_for_each(boost::irange(1u, smp::count), [this, &sharded_db, keyspace_name, ring_version] (unsigned shard) -> future<> {
+            _erms[shard] = co_await sharded_db.invoke_on(shard, [keyspace_name, ring_version] (const replica::database& db) {
+                const auto& ks = db.find_keyspace(keyspace_name);
+                auto erm = ks.get_effective_replication_map();
+                auto local_ring_version = erm->get_token_metadata().get_ring_version();
```

Comment:
Can we 100% sure that all the share will get the same ring version at any point in time even during the erm is replicated to all shards?

Reply:
Yes.  This is the way the token_metadata + replication maps are updated on all shards.
We first calculate and update the copy on shard 0 and then clone them onto all shards, including their ring version, under the shared_token_metadata lock.  Hence the on_internal_error here.

Reply:
I added a comment in v3 describing the mechanism.